### PR TITLE
FlyteRemote fetch workflow->subworkflow->launchplan pattern

### PR DESCRIPTION
**Pull request by wild-endeavor**
Signed-off-by: Yee Hing Tong <wild-endeavor@users.noreply.github.com>

# TL;DR
User reported that fetching a workflow that calls a subworkflow that calls a launch plan does not work in FlyteRemote.

The code that fetches launch plans used in a workflow was not considering subworkflows, this has been amended.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [ ] Any pending items have an associated Issue

### CODE REVIEW

File: flytekit/remote/remote.py [hunk 1]
```diff
@@ -231,12 +231,18 @@ def fetch_workflow(
         admin_workflow = self.client.get_workflow(workflow_id)
         compiled_wf = admin_workflow.closure.compiled_workflow
 
+        wf_templates = [compiled_wf.primary.template]
+        wf_templates.extend([swf.template for swf in compiled_wf.sub_workflows])
+
         node_launch_plans = {}
         # TODO: Inspect branch nodes for launch plans
```
Comment (eapolinario) [inline flytekit/remote/remote.py:238]: can we remove this TODO?

File: tests/flytekit/unit/remote/test_with_responses.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        os.path.join(responses_dir, "admin.launch_plan_pb2.LaunchPlan_core.control_flow.subworkflows.leaf_subwf.pb"),
+    )
+    leaf_lp = launch_plan_models.LaunchPlan.from_flyte_idl(leaf_lp)
+    root_wf = load_proto_from_file(
+        workflow_pb2.Workflow,
+        os.path.join(responses_dir, "admin.workflow_pb2.Workflow_core.control_flow.subworkflows.root_level_wf.pb"),
+    )
+    root_wf = admin_workflow_models.Workflow.from_flyte_idl(root_wf)
+
+    mock_client.get_workflow.side_effect = [root_wf, None]
```
Comment (eapolinario) [inline tests/flytekit/unit/remote/test_with_responses.py:34]: why is this called twice?

File: flytekit/remote/remote.py (hunk 1 above)
Comment (wild-endeavor) [reply flytekit/remote/remote.py:238]: but then i'll forget.

Comment (eapolinario) [reply flytekit/remote/remote.py:238]: oh, it's branch nodes. nvm.

**Review by eapolinario — APPROVED** [review]

**Pull request merged**